man/readtags.1.rst.in

```diff
@@ -18,54 +18,236 @@ SYNOPSIS
+
+About the ``-E`` option: certain characters are escaped in a tags file, to make
+it machine-readable. e.g., ensuring no tabs character appear in fields other
+than the pattern field. By default, readtags translates them to make it
+human-readable, but when utilizing readtags output in a script or a client
+tool, ``-E`` option should be used. See ctags-client-tools(7) for more
+discussion on this.
+
+Filtering and Sorting
+~~~~~~~~~~~~~~~~~~~~~
+Further filtering and sorting on the tags listed by actions are performed using:
+
+``-Q EXP``, ``--filter EXP``
+	Filter the tags listed by ACTION with EXP before printing.
+
+``-S EXP``, ``--sorter EXP``
+	Sort the tags listed by ACTION with EXP before printing.
+
+These are discussed in the `EXPRESSION`_ section.
+
+Examples
+~~~~~~~~
+* List all tags in "/path/to/tags":
+
+  .. code-block:: console
+
+     $ readtags -t /path/to/tags -l
+
+* List all tags in "tags" that start with ``ctags``:
+
+  .. code-block:: console
+
+     $ readtags -p - ctags
+
+* List all tags matching "ctags", case insensitively:
+
+  .. code-block:: console
+
+     $ readtags -i - ctags
+
+* List all tags start with "ctags", and printing all fields (i.e., the whole line):
+
+  .. code-block:: console
+
+     $ readtags -p -ne - ctags
 
 EXPRESSION
 ----------
+Scheme-style expressions are used for the ``-Q`` and ``-S`` options. For those
+who doesn't know Scheme or Lisp, just remember:
 
-...
+* A function call is wrapped in a pair of parenthesis. The first item in it is
+  the function/operator name, the others are arguments.
+* Function calls can be nested.
+
+So, ``(+ 1 (+ 2 3))`` means add 2 and 3 first, then add the result with 1.
 
 Filtering
 ~~~~~~~~~
+The tag entries that makes the filter expression produces non-#f values are
+filtered out (#f means false).
+
+The basic operators for filtering are ``eq?``, ``prefix?``, ``suffix?`` and
+``substr?``. Language common fields can be accessed using ``$``, e.g.,
+``$language`` represents the language field. For example:
+
+* List all tags start with "ctags" in Python code files:
+
+  .. code-block:: console
 
-...
+     $ readtags -p -Q '(eq? $language "Python")' - ctags
+
+``downcase`` or ``upcase`` operators can be used to perform case-insensitive
+matching:
+
+* List all tags containing "ctags", case insensitively:
+
+    .. code-block:: console
+
+     $ readtags -Q '(substr? (downcase $name) "ctags")' -l
+
+We have logical operators like ``and``, ``or`` and ``not``. The value of a
+missing field is #f, so we could deal with missing fields:
+
+* List all tags containing "ctags" in Python code files, but allow the
+  ``language:`` field to be missing:
+
+  .. code-block:: console
+
+     $ readtags -Q '(and (substr? $name "ctags")\
+                         (or (eq? $language "Python")\
+                             (not $language)))' -l
+
+The ``member`` operator is used to deal with lists. Currently lists can't be
```

Comment:
We plan to replace the combination of "member" and "$inherits" with regular expression matching.
So I would like you to warn readers that this one will not work in the future (, or something).

Reply:
Done.

Reply:
Thank you. Everytime you have to run `make QUICK=1 update-docs` after modifying.

Reply:
Ah yes. Somehow I committed before I ran it.